qt/applications/workbench/workbench/plotting/figuremanager.py

```diff
@@ -123,6 +124,23 @@ def replaceHandle(self, _, workspace):
         if redraw:
             self.canvas.draw_idle()
 
+    @_catch_exceptions
+    def renameHandle(self, oldName, newName):
+        """
+        Called when the ADS has renamed a workspace.
+        If this workspace is attached to this figure then the figure name is updated
+        :param oldName: The old name of the workspace.
+        :param newName: The new name of the workspace
+        """
+        for ax in self.canvas.figure.axes:
+            if isinstance(ax, MantidAxes):
+                try:  # matrix workspaces
```

Comment:
This feels like a misuse of exceptions, is it possible to query the workspace type? (e.g. via `isinstance`)